google-summer-of-code/GSOC-2022-Ideas.md

```diff
@@ -297,3 +297,40 @@ The best way to get started is to join our Slack community: https://join.slack.c
+
+The model pipeline involves predicting deforestation using Landsat 7 satellite images from the Google Earth Engine (GEE) and deforestation labels extracted from relevant datasets (such as the [matt hansen dataset](<matt hansen dataset link>)).
+
+The goal of the of this project is to *make the model available to a broader audience.*One of our goals since the inception of the project has been to facilitate decision makers to make informed decisions using our model. Open sourcing the code and creating convenient interfaces to interact with the model are steps toward this goal.
+
+| Category              | Rating                                   |
+| --------------------- | ---------------------------------------- |
+| Intensity             | Moderate                                 |
+| Priority              | Medium                                   |
+| Skills                | Python, JavaScript, UI/UX, Design Systems|
+| Project Size          | Large(350 hours)                         |
+| Preferred Contributor | Student                                  |
+| Mentors               | @govarun                                 |
+
+## Project goals
+
+- The project involves creating a web based application to interact with the model. This would involve taking as input geo-spatial coordinates from the user and connecting with the remote model server to process the query. 
+- The relevant model outputs, that is the deforestation labels for the queried region, deforestation plots, explainiablity information, have to be returned to the user on a neat and intuitively designed webpage. 
+- Since the response from the model might not be instantaneous, the frontend code should implement the functionality of a loading view while waiting for the backend. 
+- The design of the UI should be scalable to introduce new input and output parameters with minimal effort/change in the UI code-base.
+
+
+## Technical skills
+
+Experience in implementing websites and connecting them to a backend python server is a must. Understanding of Python is needed too. Otherwise the choice of UI library/tools is flexible and will be based on the consensus of the mentor and the mentee. An understanding of how a complete UI stack is setup would help in the project being finished smoothly. Knowledge of the formats in which large geographical images are stored/rendered would be preferred but it's not necessary. 
```

Comment:
Do you have any particular Python web framework in mind? Like Flask, FastAPI or Django?